### DISCUSSION TITLE
How do I set an existing join table in a many to many relationship?

### DISCUSSION THREAD

**Discussion by geoguide** [prisma/prisma#3430] [Q&A] [answered]
I'm trying to implement Prisma for an existing project. I keep running into walls that I eventually figure out but this is really annoying me. I have existing tables including an existing join table. I want to be able to do

```ts
const allUsers = await prisma.user.findMany({
    select: {
      objectId: true,
      firstName:true,
      facilities: true,
    }
  })
```

Using:

```prisma
model User {
  objectId      String @id
  firstName     String
  facilities    Facility[] @relation("UserFacility")
}

model UserFacility {
  objectId      String
  userId        User @relation(fields: [user], references: [objectId])
  user          String
  facilityId    Facility @relation(fields: [facility], references: [objectId])
  facility      String
  @@id([user, facility])
}

model Facility {
  @@map(name: "EnteredFacility") // Had to do this because table name doesn't match field name
  objectId       String @id
  center         String
  department     String
  users          User[] @relation("UserFacility")
}
```

But I keep getting `TableDoesNotExist { table: "public._UserFacility" } })`

How do I get prisma to not automatically add the underscore? Is this just a completely wrong pattern for this use case?

I got it to work changing `facilities` to `UserFacility[]` but then I just get the entries there which I don't care about. Any help would get me going a lot faster. thanks.

**Comment by nikolasburk** [comment, marked as answer]
Hey @geoguide 👋  it seems like you want to use an [explicit m-n-relation](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-schema/relations#explicit-many-to-many-relations) (explicit means that you're using a JOIN table in this case) but currently you're not following that pattern quite properly yet. In fact, I'm surprised the Prisma schema you posted actually works since it's missing the relation fields pointing from `User` and from `Facility` to `UserFacility`.

Here's how you properly model the explicit m-n relation with a JOIN table:

```prisma
model User {
  objectId       String         @id
  firstName      String
  userFacilities UserFacility[]
}

model UserFacility {
  objectId   String
  userId     User     @relation(fields: [user], references: [objectId])
  user       String
  facilityId Facility @relation(fields: [facility], references: [objectId])
  facility   String
  @@id([user, facility])
}

model Facility {
  @@map(name: "EnteredFacility")// Had to do this because table name doesn't match field name
  objectId       String         @id
  center         String
  department     String
  userFacilities UserFacility[]
}
```

To clarify: The JOIN table `UserFacility` basically has two 1-n relations: `User` <> `UserFacility` and `Facility` <> `UserFacility`. Combined, this lets you stored m-n relations because each row in the `UserFacility` table represents one relation.

As for querying this with Prisma Client, you can formulate your query like so:

```ts
const allUsers = await prisma.user.findMany({
  select: {
    objectId: true,
    firstName: true,
    userFacilities: {
      select: {
        facility: true
      }
    }
  }
})
```

Let me know if that helps or if you have any further questions! I'd definitely also recommend you to give the [Relations](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-schema/relations) page in the docs a read since a lot of the concepts are explained there 😄

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
